src/contracts/libraries/GPv2TradeExecution.sol

```diff
@@ -9,6 +9,10 @@ import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
 library GPv2TradeExecution {
     using SafeERC20 for IERC20;
 
+    /// @dev Ether marker address used to indicate an order is buying Ether.
+    address internal constant BUY_ETH_ADDRESS =
+        0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
```

Comment:
Is there a pattern to this madness?

Reply:
Yes! https://github.com/ethereum/EIPs/blob/master/EIPS/eip-55.md

Reply:
Out of curiosity, is there any reason for using this instead of 0 bytes (marginally cheaper calldata)?

Reply:
Great question! It was mostly done because of a few small reasons:
- `address(0)` is sometimes conflated with "no address" (for example `IdToAddressBiMap` uses this to indicate an empty slot)
- things like reading past the end of the calldata or memory returns `0`'s

So, all in all, I thought it would be a little safer to use a special non-zero value instead of a 0 one. Additionally, since use use token indices instead of addresses for orders, `0xeeee...eeee` will show up one over `0x0000..0000` for a total of `240` additional gas per settlement.

I think a good compromise would be to use something `0xee000000ee000000ee00000...` instead so that there are more 0 bytes and reduces calldata gas cost (this particular example will only be 60 gas more than `address(0)`).

@fleupold do you think we should change this?

Reply:
Thanks for the clarification. I don't think we should change this.

Reply:
Also people use the address 1, sometimes cause contracts do the not-zero-address assert. I've used it myself to burn once :) 

https://etherscan.io/address/0x0000000000000000000000000000000000000001

These addresses are part of some special addresses in ethereum, so I think they can never be owned, even if someone manage to get the PK (not that would be a concern to drive the decision)

Anyways, I'm fine with eeEEeeEEEeeEEEeh! one, not sure how much cheaper it is to use a the 0x1 one

Reply:
I don't think these addresses are treated any differently (maybe I'm missing something), but the cryptographic assumption of EC keys that it is close to impossible to derive a private key that results in 0x00...00 (or any chosen address for that matter) makes it safe to use.

Reply:
> These addresses are part of some special addresses in ethereum

Any address you pick without knowing the private key is impossible to own :smile:. I asked @fedgiac once if it was possible that a key exists such that its address is `0` and his answer was "maybe" in that, because of how addresses are derived, it is possible that such a key exists, but like for all addresses, finding that key is impossible (correct me if I'm wrong).

> not sure how much cheaper it is to use a the 0x1 one

It would be 228 gas cheaper per settlement (*not* order) that trades ETH.

Reply:
> I don't think these addresses are treated any differently (maybe I'm missing something), but the cryptographic assumption of EC keys that it is close to impossible to derive a private key that results in 0x00...00 (or any chosen address for that matter) makes it safe to use.

Yes, I'm not concern about the security in this. My comment was just that 0x1 looked also as good as the other one. But I'm good with any address you choose. 

With special address I meant, they are used internally by the EVM to perform some operations, so I don't think even with the PK u can extract the money there (i might be wrong though). Anyways, it wouldn't matter, if someone has that PK is cause it cracked any address anyways (visitors from another planet with super technology 👽):
https://medium.com/mycrypto/ethereums-istanbul-fork-technical-explanation-48a3aef718b0

Reply:
![image](https://user-images.githubusercontent.com/2352112/104621507-7ac5e100-5690-11eb-8b59-4cc5c8ab5b80.png)

Reply:
That's very interesting. It looks like you cannot even send funds to these addresses: https://rinkeby.etherscan.io/tx/0x9c960f4b4b96afa267b2ee48186e5babff3cc127f5ba31d9610f4464b50ab180

Reply:
Mm, at least from a contract I think we can. At least I've burned tokens like that in the past..